frontend/src/pages/Register.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+
+import React, { useState } from "react";
+import ApiPost from "../components/ApiPost";
+
+const Register = () => {
+  const [username, setUsername] = useState();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState();
+  return (
+    <div>
+      <h1>Register</h1>
+
+      <form
+        onSubmit={e => {
+          e.preventDefault();
+          e.persist();
+          ApiPost("users/new", false, { username, email, password }).then(
+            res =>
+              // After registering, show verify email page.
+              (window.location.href = "/verify-email?email=" + email)
```

Comment:
reach router has a navigate function. I believe using window.location.href will cause a full refresh of react components so everything gets mounted again. Using navigate will only unmount and mount the required components.